lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -273,7 +273,13 @@ public function bindValue($param, $value, int $type = ParameterType::STRING) : v
      */
     public function bindParam($param, &$variable, int $type = ParameterType::STRING, ?int $length = null) : void
     {
-        $param = $this->_paramMap[$param];
+        if (is_int($param)) {
+            if (! isset($this->_paramMap[$param])) {
+                throw new OCI8Exception(sprintf('Could not find variable mapping with index %d, in the SQL statement', $param));
```

Comment:
The idea behind throwing an exception is that in this case, it wouldn't have worked previously anyways, so we're just making the error more explicit. Need to double-check that before merging.

Reply:
any news?

Reply:
@eisberg I forgot about it. Thank you for the reminder.

This is a valid code example that works on `2.9` and a build from this PR:

```php
$stmt = $conn->prepare('SELECT :p FROM DUAL');
$stmt->bindValue('p', 'foo');
$stmt->execute();
echo $stmt->fetchColumn(), PHP_EOL;
// foo
```

This is an example of invalid code that is not supposed to work:

```php
$stmt = $conn->prepare('SELECT :p FROM DUAL');
$stmt->bindValue(1, 'foo');
$stmt->execute();
echo $stmt->fetchColumn(), PHP_EOL;
```

On `2.9` it produces:
> Warning: oci_bind_by_name(): ORA-01036: illegal variable name/number in lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php on line 287
> Uncaught Doctrine\DBAL\Driver\OCI8\OCI8Exception: ORA-01008: not all variables bound

On a build from this patch it produces:
> Uncaught Doctrine\DBAL\Driver\OCI8\OCI8Exception: Could not find variable mapping with index 1, in the SQL statement

Which is perfectly fine. DBAL prevents a knowingly invalid operation on the underlying driver.

/cc @beberlei